src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -956,6 +981,20 @@ private static JsonSerializerOptions GetOrCreateDefaultOptionsInstance()
             return Interlocked.CompareExchange(ref s_defaultOptions, options, null) ?? options;
         }
 
+        [RequiresUnreferencedCode(JsonSerializer.SerializationUnreferencedCodeMessage)]
+        [RequiresDynamicCode(JsonSerializer.SerializationRequiresDynamicCodeMessage)]
+        private static JsonSerializerOptions GetOrCreateWebOptionsInstance()
+        {
+            var options = new JsonSerializerOptions
+            {
+                _propertyNameCaseInsensitive = true,
+                _jsonPropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                _numberHandling = JsonNumberHandling.AllowReadingFromString
+            };
+
+            return Interlocked.CompareExchange(ref s_webOptions, options, null) ?? options;
```

Comment:
```suggestion
            return Interlocked.CompareExchange(ref s_webOptions, options, null) ?? s_webOptions;
```
That way, the property always returns the same published instance.